spec/runner_spec.rb

```diff
@@ -16,11 +16,37 @@ def on_message(&block)
   def process(message)
     @messages << message
 
-    processor = @processor_queue.shift || proc {} 
+    processor = @processor_queue.shift || proc {}
     processor.call(message)
   end
 end
 
+class TestBatchConsumer < Racecar::Consumer
+  attr_reader :messages
+
+  def initialize
+    @messages = []
+    @processor_queue = []
+  end
+
+  def on_message(&block)
+    @processor_queue << block
+    self
+  end
+
+  def process_batch(messages)
+    @messages += messages
```

Comment:
You're absolutely right about it - I didn't notice that until a bit later, after I created this PR. Will address this and come back with some updates.

Thanks!